tasmota/xdrv_17_rcswitch.ino

```diff
@@ -51,22 +51,21 @@ uint32_t rf_lasttime = 0;
 void RfReceiveCheck(void) {
   if (mySwitch.available()) {
 
-    unsigned long data = mySwitch.getReceivedValue();
+    unsigned long long data = mySwitch.getReceivedValue();
     unsigned int bits = mySwitch.getReceivedBitlength();
     int protocol = mySwitch.getReceivedProtocol();
     int delay = mySwitch.getReceivedDelay();
 
-    AddLog(LOG_LEVEL_DEBUG, PSTR("RFR: Data 0x%lX (%u), Bits %d, Protocol %d, Delay %d"), data, data, bits, protocol, delay);
+    char stemp[32];
+    ext_snprintf_P(stemp, sizeof(stemp), PSTR("\"0x%_X\""), &data);
+    AddLog(LOG_LEVEL_DEBUG, PSTR("RFR: Data 0x%s, Bits %d, Protocol %d, Delay %d"), stemp, bits, protocol, delay);
```

Comment:
You can use `0x%_X` directly in AddLog, it calls ext_snprintf_P behind the scenes

Reply:
Yes, but we may need also the Hex 64 Bits later for the JSON.
So I think it is more efficient to do it once.
Let me know @s-hadinger